src/Tester/GenericGrainTests.cs

```diff
@@ -299,7 +299,7 @@ public static void MyClassCleanup()
             await grain.SetB(b);
 
             Task<string> stringPromise = grain.GetAxB();
-            Assert.AreEqual(expected, stringPromise.Result);
```

Comment:
Then, as mentioned in the description, we should probably not rely on the TCS to sometimes avoid the deadlock, a more appropriate fix for that is probably to have the message pump to make sure to run the callbacks in a different thread that is pumping messages, otherwise it is very prone not just to deadlocks, but to prevent the message pumping to continue... If you want, I can submit another PR so that you evaluate what I think should be done in the OutsideRuntimeClient